docs/antrea-network-policy.md

```diff
@@ -1843,3 +1849,63 @@ Similar RBAC is applied to the ClusterGroup resource.
   This is due to kube-proxy performing SNAT, which conceals the original source IP from
   Antrea. Consequently, NetworkPolicies are unable to differentiate between hairpin
   Service traffic and external traffic in this scenario.
+
+### Limitations of Antrea policy logging
+
+Antrea policy logging is enabled by setting `enableLogging` to true for specific
+policy rules (or by using the `networkpolicy.antrea.io/enable-logging: "true"`
+annotation for K8s NetworkPolicies). Starting with Antrea v1.13, logging is
+"best-effort": if too much traffic needs to be logged, we will skip logging
+rather than start dropping packets or rather than risking to overrun the Antrea
+Agent, which could impact cluster health or other workloads. This behavior
+cannot be changed, and the logging feature is therefore not meant to be used for
+compliance purposes. By default, the Antrea datapath will send up to 500 packets
+per second (with a burst size of 1000 packets) to the Agent for logging. This
+rate applies to all the traffic that needs to be logged, and is enforced at the
+level of each Node. A rate of 500 packets per second roughly translates to 500
+new TCP connections per second, or 500 UDP requests per second. While it is
+possible to adjust the rate and burst size by modifying the `packetInRate`
+parameter in the antrea-agent configuration, we do not recommend doing so. The
+default value was set to 500 after careful consideration.
+
+#### Logging prior to Antrea v1.13
+
+Prior to Antrea v1.13, policy logging was not best-effort. While we did have a
+rate limit for the number of packets that could be sent to the Agent for
+logging, the datapath behavior was to drop all packets that exceeded the rate
+limit, as opposed to skipping the logging and applying the specified policy rule
+action. This meant that the logging feature was more suited for audit /
+compliance applications, however, we ultimately decided that the behavior was
+too aggressive and that it was too easy to disrupt application workloads by
+enabling logging - the rate limit was also lower than the default one we use
+today (100 packets per second instead of 500). For example, the following policy
+which allows ingress DNS traffic for coreDNS Pods, and has logging enabled,
+would drastically restrict the number of possible DNS requests in the cluster,
+which in turn would cause a lot of errors in applications which rely on DNS:
+
+```yaml
```

Comment:
Would it be more helpful to add a link to [ACNP with log settings example](https://github.com/antrea-io/antrea/blob/b3328562b39eff83bcafa2e1132e656ac740974e/docs/antrea-network-policy.md#acnp-with-log-settings) here for comparison? And maybe change that previous example to a more suitable example to log (`Drop`), as opposed to this example?

Reply:
Sorry I am having a hard time understanding what you are suggesting. What difference are we trying to highlight by comparing with the `acnp-with-log-setting` example? And which example would you want to update to use the `Drop` action (and why)?

Reply:
I assumed that `acnp-with-log-setting` example does not suffer much from disrupting application workloads, rather than this `allow-dns` example, so I was referring to this difference.

I was wondering if we need to change `acnp-with-log-setting` to `Drop` as mentioned below _especially when the policy rule uses the `Allow` action_. (Quick question is it no longer the case that only the first packet of an Allow connection will be logged?)

Reply:
> I assumed that acnp-with-log-setting example does not suffer much from disrupting application workloads, rather than this allow-dns example, so I was referring to this difference.

This policy applies to traffic from the application frontend to the DB layer. If it is a large scale application with a high number of connections, it could suffer from the same issue. In practice, for this specific case, the number of connections between the frontend and the DB is likely to stay "small", as the application is likely to use a connection pool instead of creating a new connection for each user session. But that really depends on the type of application.

The difference between these policies is more about the workloads to which they apply. I don't think there is a difference in how they are implemented. So they can both suffer from this issue in the same way.

In practice, users are more likely to experience this issue with CoreDNS because: 1) it seems common to enable logging for DNS requests, 2) even medium clusters usually have a large volume of DNS requests.

So I may still be missing your point.

> I was wondering if we need to change acnp-with-log-setting to Drop as mentioned below especially when the policy rule uses the Allow action. (Quick question is it no longer the case that only the first packet of an Allow connection will be logged?)

With recent versions of Antrea (since v1.13), logging is fine with the `Allow` action, so why should we change the example? _especially when the policy rule uses the `Allow` action_ is because the side effect of enabling logging with older Antrea versions is that flows are dropped after a certain rate limit. If the action is `Allow`, this is clearly a bigger deal than if the action is `Drop` anyway.

We only log the first packet of an `Allow` connection. That has not changed.

Reply:
Thanks for the detailed explanations! Now I understand, no need for difference, and _If the action is Allow, this is clearly a bigger deal than if the action is Drop anyway._ makes total sense.